### DISCUSSION TITLE
SSH on flux create source git fails

### DISCUSSION THREAD

**Discussion by matty1979** [fluxcd/flux2#973] [Q&A]
Following the documentation on [https://toolkit.fluxcd.io/cmd/flux_create_source_git/](flux create source)
I'm at the point of 
```bash
flux create source git monitoring \
--url=https://github.com/<repo_name>/fluxv2 \
--branch=master
```
I've tried this with `ssh:ssh@github.com/` as well 

Both come back with the same error 
`error: ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain`

I've added the generated key when it creates as a deploy key inside the github repo. 

I receive the same error using the public repo provided of fluxcd/flux2. 
I'm quite unsure as to what I'm doing wrong here.

**Comment by matty1979** [comment]
On the public repo provided, in order to make it work I had to remove --branch=main as it was giving me 
`error: couldn't find remote ref "refs/heads/master"` 

This at least is giving me some headway, but we want to use a private repo so we can include things like cert manager and nginx etc.

**Comment by suryapandian** [comment]
was this fixed?

**Comment by stefanprodan** [comment]
We changed the default branch to `main`
